share/getters/shrex.go

```diff
@@ -79,6 +136,7 @@ func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.Ex
 				"hash", root.String(),
 				"err", getErr,
 				"finished (s)", time.Since(start))
+			sg.metrics.recordEDSAttempt(ctx, attempt, false)
```

Comment:
the attempt count is only recorded once, before returning (since that was the amount of attempts necessary before either giving up or returning successfully)